pontoon/sync/vcs/models.py

```diff
@@ -37,6 +40,37 @@
 log = logging.getLogger(__name__)
 
 
+class DownloadTOMLParser(TOMLParser):
+    """
+    This wrapper is a workaround for the lack of the shared and persistent filesystem
+    on Heroku workers.
+    Related: https://bugzilla.mozilla.org/show_bug.cgi?id=1530988
+    """
+
+    def __init__(self, checkout_path, permalink_prefix):
+        self.checkout_path = checkout_path
+        self.permalink_prefix = permalink_prefix
+
+    def parse(self, path, env=None, ignore_missing_includes=True):
+        config_path = path.replace(self.checkout_path, "")
+
+        if "{locale_code}" in self.permalink_prefix:
+            remote_path = (self.permalink_prefix or "").format(locale_code=config_path)
+        else:
+            remote_path = urljoin(self.permalink_prefix or "", config_path)
```

Comment:
I can't see how this would work if `self.permalink_prefix` is None. If it is, I think we should raise an error.